InsightsFrontend/src/app/pages/release-roadmap/release-roadmap.component.ts

```diff
@@ -326,15 +359,33 @@ export class ReleaseRoadmapComponent implements OnInit, AfterViewInit {
 
   private sortMilestones(milestones: Milestone[]): Milestone[] {
     return milestones.sort((a, b) => {
-      // First sort by due date
-      const dueComparison = (a.dueOn?.getTime() ?? 0) - (b.dueOn?.getTime() ?? 0);
-      if (dueComparison !== 0) return dueComparison;
+      const aDue = a.dueOn ? new Date(a.dueOn).getTime() : Number.MAX_SAFE_INTEGER;
+      const bDue = b.dueOn ? new Date(b.dueOn).getTime() : Number.MAX_SAFE_INTEGER;
```

Comment:
Why MAX_SAFE_INTEGER?
Wouldn't a time value of `-1` and adding a check in the places where it's needed be better?